lib/ansible/module_utils/common/parameters.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.common._collections_compat import (
+    KeysView,
+    Sequence,
+)
+
+from ansible.module_utils.six import (
+    binary_type,
+    text_type,
+)
+
+PASS_VARS = {
+    'check_mode': 'check_mode',
+    'debug': '_debug',
+    'diff': '_diff',
+    'keep_remote_files': '_keep_remote_files',
+    'module_name': '_name',
+    'no_log': 'no_log',
+    'remote_tmp': '_remote_tmp',
+    'selinux_special_fs': '_selinux_special_fs',
+    'shell_executable': '_shell',
+    'socket': '_socket_path',
+    'syslog_facility': '_syslog_facility',
+    'tmpdir': '_tmpdir',
+    'verbosity': '_verbosity',
+    'version': 'ansible_version',
+}
+
+# Note: When getting Sequence from collections, it matches with strings. If
+# this matters, make sure to check for strings before checking for sequencetype
+SEQUENCETYPE = frozenset, KeysView, Sequence
+DEFAULT_LEGAL_PARAMS = ['_ansible_%s' % k for k in PASS_VARS]
+
+
+def handle_aliases(argument_spec, params, legal_inputs=None):
+    if legal_inputs is None:
+        legal_inputs = DEFAULT_LEGAL_PARAMS[:]
+    aliases_results = {}  # alias:canon
+
+    for (k, v) in argument_spec.items():
+        legal_inputs.append(k)
+        aliases = v.get('aliases', None)
+        default = v.get('default', None)
+        required = v.get('required', False)
+        if default is not None and required:
+            # not alias specific but this is a good place to check this
+            raise ValueError("internal error: required and default are mutually exclusive for %s" % k)
+        if aliases is None:
+            continue
+        if not isinstance(aliases, SEQUENCETYPE) or isinstance(aliases, (binary_type, text_type)):
```

Comment:
Instead of moving SEQUENCETYPE, I think we can switch this to using is_iterable.

``` python
from ansible.module_utils.common.collections import is_iterable
[...]
[Line 56 becomes]
if not is_iterable(aliases):
```